tests/utils/utils.go

```diff
@@ -70,92 +73,73 @@ func ReadHexTextFile(filename string) string {
 	return strings.TrimRight(string(fileData), "\n")
 }
 
+// Constructs a relayer config with all subnets as sources and destinations
 func CreateDefaultRelayerConfig(
-	subnetAInfo interfaces.SubnetTestInfo,
-	subnetBInfo interfaces.SubnetTestInfo,
+	subnetsInfo []interfaces.SubnetTestInfo,
 	teleporterContractAddress common.Address,
 	fundedAddress common.Address,
 	relayerKey *ecdsa.PrivateKey,
 ) config.Config {
-	hostA, portA, err := teleporterTestUtils.GetURIHostAndPort(subnetAInfo.NodeURIs[0])
-	Expect(err).Should(BeNil())
+	// Construct the config values for each subnet
+	hosts := make([]string, len(subnetsInfo))
+	ports := make([]uint32, len(subnetsInfo))
+	sources := make([]config.SourceSubnet, len(subnetsInfo))
+	destinations := make([]config.DestinationSubnet, len(subnetsInfo))
+	blockchainIDs := make([]string, len(subnetsInfo))
+	subnetIDs := make([]string, len(subnetsInfo))
+	for i, subnetInfo := range subnetsInfo {
+		var err error
+		hosts[i], ports[i], err = teleporterTestUtils.GetURIHostAndPort(subnetInfo.NodeURIs[0])
+		Expect(err).Should(BeNil())
+
+		sources[i] = config.SourceSubnet{
+			SubnetID:          subnetInfo.SubnetID.String(),
+			BlockchainID:      subnetInfo.BlockchainID.String(),
+			VM:                config.EVM.String(),
+			EncryptConnection: false,
+			APINodeHost:       hosts[i],
+			APINodePort:       ports[i],
+			MessageContracts: map[string]config.MessageProtocolConfig{
+				teleporterContractAddress.Hex(): {
+					MessageFormat: config.TELEPORTER.String(),
+					Settings: map[string]interface{}{
+						"reward-address": fundedAddress.Hex(),
+					},
+				},
+			},
+		}
 
-	hostB, portB, err := teleporterTestUtils.GetURIHostAndPort(subnetBInfo.NodeURIs[0])
-	Expect(err).Should(BeNil())
+		destinations[i] = config.DestinationSubnet{
+			SubnetID:          subnetInfo.SubnetID.String(),
+			BlockchainID:      subnetInfo.BlockchainID.String(),
+			VM:                config.EVM.String(),
+			EncryptConnection: false,
+			APINodeHost:       hosts[i],
+			APINodePort:       ports[i],
+			AccountPrivateKey: hex.EncodeToString(relayerKey.D.Bytes()),
+		}
+
+		blockchainIDs[i] = subnetInfo.BlockchainID.String()
+		subnetIDs[i] = subnetInfo.SubnetID.String()
+	}
 
 	log.Info(
 		"Setting up relayer config",
-		"hostA", hostA,
-		"portA", portA,
-		"blockChainA", subnetAInfo.BlockchainID.String(),
-		"hostB", hostB,
-		"portB", portB,
-		"blockChainB", subnetBInfo.BlockchainID.String(),
-		"subnetA", subnetAInfo.SubnetID.String(),
-		"subnetB", subnetBInfo.SubnetID.String(),
+		"hosts", hosts,
+		"port", ports,
+		"blockchainIDs", blockchainIDs,
+		"subnetIDs", subnetIDs,
```

Comment:
If we move the log inside the foreach loop, we then don't need the slices for hosts, ports, blockchainIDs, subnetIDs